stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+    "m4.xlarge": 58,
+    "m4.2xlarge": 58,
+    "m4.4xlarge": 234,
+    "m4.10xlarge": 234,
+    "m5.large": 29,
+    "m5.xlarge": 58,
+    "m5.2xlarge": 58,
+    "m5.4xlarge": 234,
+    "m5.12xlarge": 234,
+    "m5.24xlarge": 737,
+    "p2.xlarge": 58,
+    "p2.8xlarge": 234,
+    "p2.16xlarge": 234,
+    "p3.2xlarge": 58,
+    "p3.8xlarge": 234,
+    "p3.16xlarge": 234,
+    "r3.xlarge": 58,
+    "r3.2xlarge": 58,
+    "r3.4xlarge": 234,
+    "r3.8xlarge": 234,
+    "r4.large": 29,
+    "r4.xlarge": 58,
+    "r4.2xlarge": 58,
+    "r4.4xlarge": 234,
+    "r4.8xlarge": 234,
+    "r4.16xlarge": 737,
+    "t2.small": 8,
+    "t2.medium": 17,
+    "t2.large": 35,
+    "t2.xlarge": 44,
+    "t2.2xlarge": 44,
+    "x1.16xlarge": 234,
+    "x1.32xlarge": 234,
+}
+
+
+def create_max_pods_per_node_mapping(t):
+    mapping = {}
+    for instance, max_pods in max_pods_per_instance.iteritems():
+        mapping[instance] = {"MaxPods": max_pods}
+    t.add_mapping("MaxPodsPerNode", mapping)
+
+
+# This is copy/pasted from
+# https://amazon-eks.s3-us-west-2.amazonaws.com/1.10.3/2018-06-05/amazon-eks-nodegroup.yaml
+# and updated to call troposphere functions instead of EC2 CFN placeholders
+def get_launch_config_userdata(cluster_name, instance_type):
+    if instance_type not in max_pods_per_instance:
+        raise ValueError("%s is not supported by EKS" % instance_type)
+
+    launch_config_userdata = [
+        "#!/bin/bash -xe\n",
+        "CA_CERTIFICATE_DIRECTORY=/etc/kubernetes/pki", "\n",
+        "CA_CERTIFICATE_FILE_PATH=$CA_CERTIFICATE_DIRECTORY/ca.crt", "\n",
+        "MODEL_DIRECTORY_PATH=~/.aws/eks", "\n",
+        "MODEL_FILE_PATH=$MODEL_DIRECTORY_PATH/eks-2017-11-01.normal.json", "\n",
+        "mkdir -p $CA_CERTIFICATE_DIRECTORY", "\n",
+        "mkdir -p $MODEL_DIRECTORY_PATH", "\n",
+        "curl -o $MODEL_FILE_PATH https://s3-us-west-2.amazonaws.com/amazon-eks/1.10.3/2018-06-05/eks-2017-11-01.normal.json",
+        "\n",
+        "aws configure add-model --service-model file://$MODEL_FILE_PATH --service-name eks", "\n",
+        "aws eks describe-cluster --region=", Ref("AWS::Region"), " --name=", cluster_name,
+        " --query 'cluster.{certificateAuthorityData: certificateAuthority.data, endpoint: endpoint}' > /tmp/describe_cluster_result.json",
+        "\n",
+        "cat /tmp/describe_cluster_result.json | grep certificateAuthorityData | awk '{print $2}' | sed 's/[,\"]//g' | base64 -d >  $CA_CERTIFICATE_FILE_PATH",
+        "\n",
+        "MASTER_ENDPOINT=$(cat /tmp/describe_cluster_result.json | grep endpoint | awk '{print $2}' | sed 's/[,\"]//g')",
+        "\n",
+        "INTERNAL_IP=$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)", "\n",
+        "sed -i s,MASTER_ENDPOINT,$MASTER_ENDPOINT,g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,CLUSTER_NAME,", cluster_name, ",g /var/lib/kubelet/kubeconfig", "\n",
+        "sed -i s,REGION,", Ref("AWS::Region"), ",g /etc/systemd/system/kubelet.service", "\n",
```

Comment:
Can you replace the Cloudformation Metaparameters with their troposphere equivalents?  Instead of `Ref("AWS::Region")` you would use:

```
from troposphere import Region
```

And use that instead - you can see an example in: https://github.com/cloudtools/stacker_blueprints/blob/master/stacker_blueprints/ecs.py#L155